Reduced craft time of construction materials

chaisawlajatang opened the pull request:
**What**

Drastically reduces craft times of quicklime, cement, mortar, concrete and bricks.

**Why**

For now, it requires 210 minutes total to make 1 mortar or 1 concrete. Each brick wall requires 2 mortar, each concrete construction requires 8-12 concrete, rendering self-made stone buildings almost impossible.
Quicklime and cement are ridiculously time-consuming (90 min per 0.75 kg!). It would make sense to reduce craft times of quicklime to 15 min (we are burning <1 kg of a soft rock) and cement to 30 min (for some chemical processes).
Mortar or concrete are made by simply mixing components, so they should be crafted 15 mins instead of 30.

In total, 1 mortar/concrete would take 1 hour.

Also, it requires 40 min to make a single brick, and you'll need 30 of them (20 hours of nonstop craft) to build a wall tile, so craft time is decreased to 10 min.

These changes would affect stone constructions only (quicklime is also used in Prussian blue tablets and glass shards), and you still need significant amount of time and resources to build something.

Mecares commented:
I personally hope the brick kiln is getting handled like the smoking rack change in the future

Night-Pryanik commented on data/json/construction.json:
```diff
@@ -889,6 +889,7 @@
         "category" : "CONSTRUCT",
         "required_skills" : [ [ "fabrication", 4 ] ],
         "time" : 180,
+		"tools": [ [ [ "con_mix", 50 ] ] ],
```
Replace tabs with spaces here and below.

chaisawlajatang commented:
Ok.

**What's changed**

 - Added batch factor to bricks - so the 1st brick goes in 40 minutes (like before), 5th+ in 10 minutes, batch of 20 bricks takes 4 hours 4 minutes 
 - Returned cement craft time back to 90 minutes, but added a batch factor, batch of 20x (1000 total) takes 8 hours 56 minutes.
 - Changed mortar and concrete recipe (mortar requires sand instead of pebble, concrete no longer requires water)
 - Changed mortar and concrete material types from '''Stone''' to '''Powder'''
 - Mortar and concrete craft take 10 minutes now
 - Added concrete mixer as a required tool in related construction recipes.

http://www.madehow.com/Volume-1/Concrete.html

> Preparing Portland cement

> The limestone, silica, and alumina that make up Portland cement are dry ground into a very fine powder, mixed together in predetermined proportions, preheated, and calcined (heated to a high temperature that will burn off impurities without fusing the ingredients). Next the material is burned in a large rotary kiln at 2,550 degrees Fahrenheit (1,400 degrees Celsius). At this temperature, the material partially fuses into a substance known as clinker.

So we need to form a clinker via fusing the components at high temperatures. 

> The clinker is then cooled and ground to a fine powder in a tube or ball mill

And we need to grind it in some way, I'm not sure how to simulate this (or should it be simulated at all?).

https://en.wikipedia.org/wiki/Cement_kiln#The_rotary_kiln

>  In contrast to static kilns, the material passes through quickly: it takes from 3 hours (in some old wet process kilns) to as little as 10 minutes (in short precalciner kilns).

It's the only information I could find about the time needed for the cement making. So we have "several hours". Maybe batch factor would make sense here too, as we just heat more material in the same forge?

About the mortar.

https://www.diy-waterproofing.co.uk/diy-advice/making-waterproof-cement-or-mortar/

> For mortar the standard ratio is: 1 part cement and 3 parts sand

> For the masonry of exterior walls with hard brick one often uses 1 part cement, 3.5 parts sand, 0.25 part lime

https://www.doityourself.com/stry/how-to-make-brick-mortar

> For simple brick mortar, the amount of lime added is actually very little, only about 10 percent of your total volume; Portland cement and sand make up almost all the mixture. The combination should be very rich and should not have a very stiff consistency so that it can be applied to bricks with relative ease. The perfect mix for the brick laying mortar contains two parts washed sand and one part Portland cement

So looks like pebble in the recipe should be replaced by the sand (plus some amount of quicklime? I'm not sure, some sources and my IRL experience say only sand and cement are used while making mortar). The ratios differ, but 50 cement+150 sand will do. 
Dry mortar is basically a powder, so it shouldn't take much time to craft - it's just mixing two powders. 
Dry concrete is also a powder, so I removed water from the requirements. Water is added right before the immediate use (already in the construction recipes), so added a concrete mixer as a tool.

Night-Pryanik commented on data/json/construction.json:
```diff
@@ -889,6 +889,7 @@
         "category" : "CONSTRUCT",
         "required_skills" : [ [ "fabrication", 4 ] ],
         "time" : 180,
+	    "tools": [ [ [ "con_mix", 50 ] ] ],
```
The tabs are still here.

Night-Pryanik commented on data/json/recipes/recipe_others.json:
```diff
@@ -3113,7 +3112,8 @@
     "difficulty": 4,
     "time": 90000,
-    "autolearn": true,
+	"autolearn": true,
```
And here.

chaisawlajatang commented:
https://www.wikihow.com/Make-Quicklime

> Your kiln needs to be able to maintain high temperatures (over 1000 degrees Celsius) for up to an hour

It's the only information I got about time needed, so I set craft time to 60 minutes plus added batch factor.

kevingranade commented:
Awesome, that's exactly the kind of thing I was hoping for.

Bricks are eventually going to need an offline processing thing, but I think a big batch works for now.

That cement recipe is gnarly. Again we need to have better support for offline processing, it looks like there are several parts where you can walk away and do something else.
> And we need to grind it in some way, I'm not sure how to simulate this (or should it be simulated at all?).

The current tool for this is the mortar and pestle, which I believe is already used elsewhere in the recipe.  For batch grinding I believe a quern and millwheel would be ideal, but that's a whole different thing too.
> Maybe batch factor would make sense here too, as we just heat more material in the same forge?

Yea, basically your batch size is determined by your container that you do mixing and heating in.  I believe you can just use the cement mixer and a heat source?  That's not ideal, but it should be good enough.

> So looks like pebble in the recipe should be replaced by the sand

I agree, if I understand correctly the lime is nice to have, I suspect the pebble was in there because we didn't have sand at the time.

> Dry mortar is basically a powder, so it shouldn't take much time to craft - it's just mixing two powders.
> Dry concrete is also a powder, so I removed water from the requirements.

I agree.

Night-Pryanik commented on data/json/construction.json:
```diff
@@ -889,6 +889,7 @@
         "category" : "CONSTRUCT",
         "required_skills" : [ [ "fabrication", 4 ] ],
         "time" : 180,
+        "tools": [ [ [ "con_mix", 50 ] ] ],
```
I think there should be two square brackets, not three, like `[ [ "con_mix", 50 ] ]`.

ZhilkinSerg commented on data/json/recipes/recipe_others.json:
```diff
@@ -3095,7 +3095,7 @@
     "difficulty": 3,
     "time": 60000,
-    "batch_time_factors": [75, 4], 
+    "batch_time_factors": [ 75, 4 ], 
```
There is stray whitespace after comma here and below.
